extensions/mssql/src/hdfs/ui/hdfsManageAccessDialog.ts

```diff
@@ -246,7 +246,7 @@ export class ManageAccessDialog {
 		}
 
 		// If this is the first time go through and create the UI components now that we have a model to use
-		if (!this.viewInitialized) {
+		if (!this.viewInitializedOrInProgress) {
```

Comment:
Hrm, this is still potentially problematic. It's possible that the initializing is in progress but we haven't actually created all the components yet. In which case we may get a null ref trying to access those elements (such as the stickyCheckbox below). 

I think the best fix is to do what I did for the BDC and queue up stuff to be ran after initialization - https://github.com/microsoft/azuredatastudio/blob/master/extensions/big-data-cluster/src/bigDataCluster/dialog/bdcDashboardPage.ts#L32

Reply:
Ack, will make this change when I have a few free mins

Reply:
I didn't want to introduce yet another deferred implementation back into the mssql extension so I went with an EventEmitter based approach instead. Lemme know if you'd like further changes based on your BDC expertise 😄

Reply:
I like it! I have a tiny bit of a concern with a possible race condition on https://github.com/microsoft/azuredatastudio/pull/9114/files#diff-860b5cb787caedb6fb4069e8c348c314R599 where viewInitialized is false at that point, but then before it can add the event handler the event is fired and so that call is lost. But I'm not actually sure that can happen given the single-threaded nature of Javascript - as long as the calls are sequential like this it should be fine....

Reply:
@Charles-Gagnon agreed, I don't think that it can happen.